.github/workflows/docker-build-and-push.yaml

```
@@ -0,0 +1,34 @@
+name: Build and Push Docker Image
+on:
+  push:
+    branches: [ main ]
+   
+jobs:
+  build-and-push:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
```

actionlint (1.7.4)

the runner of "actions/checkout@v3" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)